lib/data/providers/router_provider.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:go_router/go_router.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import '../../constants/routes.dart';
+import '../../presentation/feature/detail/detail_page.dart';
+import '../../presentation/feature/home/home_page.dart';
+
+part 'router_provider.g.dart';
+
+@riverpod
+GoRouter goRouter(ref) => GoRouter(
+      routes: <GoRoute>[
+        GoRoute(
+          path: RoutePaths.home,
+          builder: (context, state) {
+            return const HomePage();
+          },
+          routes: [
+            GoRoute(
+              name: RouteNames.animeDetail,
+              path: RoutePaths.animeDetail,
+              builder: (context, state) {
+                return DetailPage(
```

Comment:
can we simply using path not the name?

ref - https://pub.dev/packages/go_router/example 

        ```
  GoRoute(
            path: 'page2',
            builder: (BuildContext context, GoRouterState state) =>
                const Page2Screen(),
          ),
        ],

```